pkg/util/minio.go

```diff
@@ -16,10 +16,18 @@ const (
 )
 
 var (
-	MinioClient *minio.Client
+	minioClient *minio.Client
 )
 
-func InitMinio() (err error) {
+func GetMinioClient() (*minio.Client, error) {
+	var err error
+	if minioClient == nil {
+		err = initMinio()
+	}
+	return minioClient, err
```

Comment:
It seems `GetMinioClient` function calls in three async places, but there is no synchronization primitive to protect `minioClient`. On the other hand, is it necessary to use a Single-Instance pattern in here?